File: app/src/main/java/com/sanislo/movieapp/domain/video/VideoRepository.java
package com.sanislo.movieapp.domain.video;

import android.arch.lifecycle.LiveData;

import com.sanislo.movieapp.domain.model.VideoModel;
import com.sanislo.movieapp.domain.model.YoutubeVideoModel;
import com.sanislo.movieapp.persistence.response.movieVideos.MovieVideosResponse;

import java.util.List;

import io.reactivex.Single;

public interface VideoRepository {
    Single<MovieVideosResponse> videos(int movieId);
    LiveData<List<VideoModel>> videosLiveData(int movieId);
    LiveData<List<YoutubeVideoModel>> youtubeVideosLiveData(int movieId);
}

File: app/src/main/java/com/sanislo/movieapp/domain/video/VideoRepository.kt
package com.sanislo.movieapp.domain.video

import androidx.lifecycle.LiveData

import com.sanislo.movieapp.domain.model.VideoModel
import com.sanislo.movieapp.domain.model.YoutubeVideoModel
import com.sanislo.movieapp.persistence.response.movieVideos.MovieVideosResponse

import io.reactivex.Single

interface VideoRepository {
    fun videos(movieId: Int): Single<MovieVideosResponse>
    fun videosLiveData(movieId: Int): LiveData<List<VideoModel>>
    fun youtubeVideosLiveData(movieId: Int): LiveData<List<YoutubeVideoModel>>
}
